src/test/java/com/example/weather/client/mappers/WeatherDataMapperTest.java

```diff
@@ -1,45 +1,89 @@
 package com.example.weather.client.mappers;
 
-import com.example.weather.client.models.dto.WeatherDataDto;
-import com.example.weather.client.models.dto.WeatherDto;
-import com.example.weather.client.models.entity.Weather;
-import com.example.weather.client.models.entity.WeatherData;
+import com.example.weather.client.models.dto.*;
+import com.example.weather.client.models.entity.*;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
 import static com.example.weather.client.utility.PodamUtility.makePojo;
-import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
 
+@ExtendWith(MockitoExtension.class)
 class WeatherDataMapperTest {
 
-    private WeatherDataDto weatherDataDto = makePojo(WeatherDataDto.class);
-    private final WeatherDataMapper weatherDataMapper = new WeatherDataMapper(
-            new GeoCoordinatesMapper(),
-            new WeatherMapper(),
-            new ConditionsMapper(),
-            new WindMapper());
+    private WeatherDataDto expectedWeatherDataDto = makePojo(WeatherDataDto.class);
+    private WeatherData expectedWeatherData = makePojo(WeatherData.class);
+    private GeoCoordinates geoCoordinates = makePojo(GeoCoordinates.class);
+    private Weather weather = makePojo(Weather.class);
+    private Conditions conditions = makePojo(Conditions.class);
+    private Wind wind = makePojo(Wind.class);
+    private GeoCoordinatesDto geoCoordinatesDto = makePojo(GeoCoordinatesDto.class);
+    private WeatherDto weatherDto = makePojo(WeatherDto.class);
+    private ConditionsDto conditionsDto = makePojo(ConditionsDto.class);
+    private WindDto windDto = makePojo(WindDto.class);
+
+    @Mock
+    private GeoCoordinatesMapper geoCoordinatesMapper;
+    @Mock
+    private WeatherMapper weatherMapper;
+    @Mock
+    private ConditionsMapper conditionsMapper;
+    @Mock
+    private WindMapper windMapper;
+    @InjectMocks
+    private WeatherDataMapper weatherDataMapper;
 
     @Test
     void testToEntity() {
-        var weatherData = weatherDataMapper.toEntity(weatherDataDto);
+        when(geoCoordinatesMapper.toEntity(any(GeoCoordinatesDto.class)))
+                .thenReturn(geoCoordinates);
+        when(weatherMapper.toEntity(any(WeatherDto.class)))
+                .thenReturn(weather);
+        when(conditionsMapper.toEntity(any(ConditionsDto.class)))
+                .thenReturn(conditions);
+        when(windMapper.toEntity(any(WindDto.class)))
+                .thenReturn(wind);
```

Comment:
I think it's redundant, you could simply not check those values with assertions. However it can stay, shouldn't hurt us.